### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by aekong**
Review site here: http://ci-66-sfdesignsystem.pantheonsite.io

Latest redesign deprecates a lot of the currently existing patterns, so using this opportunity to remove a bunch of stuff, most of it was a straight port from pattern lab out of sf.gov.  Also, trying to make this pattern library more generic (not twig/drupal specific), so going back to handlebars.

Components are in `src/components` and each component has it's own folder with the necessary files.  The following being most common:
- xyz.hbs
- _xyz.scss
- xyz.config.yml
- xyz.js

This PR isn't ready to merge into master yet.  Its intention to get another set of eyes and to start a conversation on whether or not this is a good direction forward.

There's still a bunch of TODO's:
- remove old folder structure (`/components`, `/docs`, `/gulptasks`, `public/css`, `public/fonts`, etc)
- refactor `gulpfile.babel.js` to remove tasks that are no longer relevant
- add tests
- probably other stuff i'm not thinking of

### CODE REVIEW

**Comment by aekong** [comment]
Latest redesign deprecates a lot of the currently existing patterns, so using this opportunity to remove a bunch of stuff, most of it was a straight port from pattern lab out of sf.gov.  Also, trying to make this pattern library more generic (not twig/drupal specific), so going back to handlebars.

Components are in `src/components` and each component has it's own folder with the necessary files.  The following being most common:
- xyz.hbs
- _xyz.scss
- xyz.config.yml
- xyz.js

This PR isn't ready to merge into master yet.  Its intention to get another set of eyes and to start a conversation on whether or not this is a good direction forward.

There's still a bunch of TODO's:
- remove old folder structure (`/components`, `/docs`, `/gulptasks`, `public/css`, `public/fonts`, etc)
- refactor `gulpfile.babel.js` to remove tasks that are no longer relevant
- add tests
- probably other stuff i'm not thinking of

File: .editorconfig
Comment (jrubenoff) [inline .editorconfig]: What does this do? Can't find docs for this on editorconfig's site.

File: fractal.js
Comment (jrubenoff) [inline fractal.js]: Maybe just "San Francisco Design System?" Ideally other agencies are using it...

File: src/components/00-design-tokens/colors.hbs
Comment (jrubenoff) [inline src/components/00-design-tokens/colors.hbs]: Can you move all inline styles to `pattern-scaffolding.scss`?

**Review by jrubenoff — CHANGES_REQUESTED** [review]
Left a few comments inline.

Should I comment on CSS architecture, or is it important that the SF.gov and design system mixins stay in sync?

**Comment by aekong** [comment]
Not super important.  It'd be nice to keep variable and mixin names the same, but if there's a good reason for changes (like better practices), totally happy to make those changes.  Otherwise, css architecture is fair game, have at it!

File: .editorconfig
Comment (aekong) [reply .editorconfig]: 😬 that's a vs code config, sorry

File: src/components/00-design-tokens/_breakpoints.scss
Comment (jrubenoff) [inline src/components/00-design-tokens/_breakpoints.scss]: Breakpoints should be `rem`s so that they work even when:

- The user has an enlarged default font size
- The zoom level is over 100%
- We increase the `font-size` property in the `html` tag

File: src/components/00-design-tokens/_breakpoints.scss
Comment (jrubenoff) [inline src/components/00-design-tokens/_breakpoints.scss]: If we want to namespace these, I feel like `mq-` (media query) might be a more intuitive prefix?

File: src/components/00-design-tokens/_colors.scss
Comment (jrubenoff) [inline src/components/00-design-tokens/_colors.scss]: @aekong Do you feel like these need to be namespaced? I'm not sure we'll ever be in danger of having two conflicting `$white`s or `$blue-1`s, but maybe you feel differently

File: src/components/00-design-tokens/_mixins.scss
Comment (jrubenoff) [inline src/components/00-design-tokens/_mixins.scss]: Seems like the sole purpose of these mixins is to save you from typing a few characters. I feel like we should remove them? They do more to obfuscate the code than make us more efficient

File: src/components/00-design-tokens/_typography-mixins.scss
Comment (jrubenoff) [inline src/components/00-design-tokens/_typography-mixins.scss:2]: We need to add some `@if` statements here to incorporate Mozilla's different syntax for this property. (e.g. `-moz-osx-font-smoothing: grayscale;`)

File: src/components/00-design-tokens/_typography-mixins.scss
Comment (jrubenoff) [inline src/components/00-design-tokens/_typography-mixins.scss]: Not sure these `fs-` mixins benefit from being namespaced either?

Also, we should convert `px` units to `rem`s, for the same reasons mentioned above

File: src/components/03-containers/three-column-container/_three-column-container.scss
Comment (jrubenoff) [inline src/components/03-containers/three-column-container/_three-column-container.scss]: Since most of our components have `10px` multiples for spacing and sizing, maybe we should turn that into a `$rhythm` variable (using `rem`s) to embed that into the system?

Examples:

`margin-bottom: $rhythm;`
`padding: $rhythm * 2;`

File: src/components/03-containers/three-column-container/_three-column-container.scss
Comment (jrubenoff) [inline src/components/03-containers/three-column-container/_three-column-container.scss]: Would it be possible to convert these selectors into classes? Pretty high specificity here.

**Comment by jrubenoff** [comment]
@aekong Alright, those are all my comments.

File: src/components/03-containers/three-column-container/_three-column-container.scss
Comment (aekong) [reply src/components/03-containers/three-column-container/_three-column-container.scss]: i think we talked about this in person, but i forgot where we landed.  The reason i did it here in css was so that it wouldn't need to add a `third-child` class in the markup, which would require some calculation

Comment (jrubenoff) [reply src/components/03-containers/three-column-container/_three-column-container.scss]: To sum up [our Slack convo:](https://sfdigitalservices.slack.com/archives/CEERV5DCG/p1573175275001200?thread_ts=1573170435.000100&cid=CEERV5DCG) it's fine to leave this selector as-is.

Might be a bit more readable if you combined lines 15 and 16, though.

File: src/components/00-design-tokens/_breakpoints.scss
Comment (aekong) [reply src/components/00-design-tokens/_breakpoints.scss]: these breakpoints are coming from figma: https://www.figma.com/file/YkshTKA1RuQEcIju7cblRR4B/Grids?node-id=0%3A1.  When I convert these to rem's (html font size is 17px) the layout breaks on the incorrect pixel width.

For example, according to the figma file, the a component goes to one column layout at 769px.  When I convert this to rem (45.24rem), the actual pixel break happens at ~732px.

Is this a common scenario?

**Review by jrubenoff — APPROVED** [review]
I made a small update to the `antialiasing` mixin.

Feel free to merge this in if that looks good to you!

**Review by aekong — COMMENTED** [review]
👍

**Pull request merged**
